wiring/inc/spark_wiring_ble.h

```diff
@@ -424,9 +427,23 @@ class BleCharacteristic {
 public:
     BleCharacteristic();
     BleCharacteristic(const BleCharacteristic& characteristic);
-    BleCharacteristic(const char* desc, EnumFlags<BleCharacteristicProperty> properties, BleOnDataReceivedCallback callback = nullptr, void* context = nullptr);
-    BleCharacteristic(const String& desc, EnumFlags<BleCharacteristicProperty> properties, BleOnDataReceivedCallback callback = nullptr, void* context = nullptr)
-            : BleCharacteristic(desc.c_str(), properties, callback, context) {
+    BleCharacteristic(EnumFlags<BleCharacteristicProperty> properties,  const char* desc, BleOnDataReceivedCallback callback = nullptr, void* context = nullptr);
```

Comment:
The context is that for the constructors that use the default service and characteristic UUID, if the callback and context are present, it is ambiguous to the template constructors with custom service and characteristic UUID while the callback and context are omitted. Surprisingly no one reported this compilation issue. I know this is a breaking change, but I assume that it would affect a very mall amount of users. And it's 3.x, it's the right time we fix this and deprecate the previous ambiguous APIs.

Reply:
And this issue is caught by the wiring API test we just introduced for BLE. Thanks to the wiring API test! 🎉

Reply:
Cool, let's make sure to add deprecation notices then, otherwise we'll never add them :)

Reply:
Will do.